homeassistant/components/amcrest/config_flow.py

```diff
@@ -0,0 +1,233 @@
+
+        data = self.discovery_schema or {
+            vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+            vol.Required(CONF_HOST): str,
+            vol.Required(CONF_USERNAME): str,
+            vol.Required(CONF_PASSWORD): str,
+            vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+        }
+
+        return self.async_show_form(
+            step_id="user",
+            description_placeholders=self.device_config,
+            data_schema=vol.Schema(data),
+            errors=errors,
+        )
+
+    async def _create_entry(self):
+        """Create entry for device."""
+
+        return self.async_create_entry(
+            title=self.device_config[CONF_NAME], data=self.device_config
+        )
+
+
+class AmcrestOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle Amcrest device options."""
+
+    def __init__(self, config_entry):
+        """Initialize Amcrest device options flow."""
+        self.config_entry = config_entry
+        self.options = dict(config_entry.options)
+        self.device = None
+
+    async def async_step_init(self, user_input=None):
+        """Manage the Amcrest device options."""
+        self.device = self.hass.data[DOMAIN][DEVICES][self.config_entry.data[CONF_NAME]]
+        return await self.async_step_configure_stream()
+
+    async def async_step_configure_stream(self, user_input=None):
+        """Manage the Amcrest device options."""
+        if user_input is not None:
+            self.options.update(user_input)
+            return self.async_create_entry(title="", data=self.options)
+
+        return self.async_show_form(
+            step_id="configure_stream",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_STREAM_SOURCE,
+                        default=self.config_entry.options.get(
+                            CONF_STREAM_SOURCE, DEFAULT_STREAM_SOURCE
+                        ),
+                    ): vol.In(STREAM_SOURCE_LIST),
+                    vol.Optional(
+                        CONF_FFMPEG_ARGUMENTS,
+                        default=self.config_entry.options.get(
+                            CONF_FFMPEG_ARGUMENTS, DEFAULT_FFMPEG_ARGUMENTS
+                        ),
+                    ): str,
+                    vol.Optional(
+                        CONF_RESOLUTION,
+                        default=self.config_entry.options.get(
+                            CONF_RESOLUTION, DEFAULT_RESOLUTION
+                        ),
+                    ): vol.All(vol.In(RESOLUTION_LIST)),
+                    vol.Optional(
+                        CONF_AUTHENTICATION,
+                        default=self.config_entry.options.get(
+                            CONF_AUTHENTICATION, DEFAULT_AUTHENTICATION
+                        ),
+                    ): vol.All(vol.In(AUTHENTICATION_LIST)),
+                    vol.Optional(
+                        CONF_CONTROL_LIGHT,
```

Comment:
Drop this option and add the light entity and let it be disabled by entity registry. Entities have the `entity_registry_enabled_default` property which you can use to disable an entity by default.

Reply:
but this is not a light entity. It is not used to turn the light on or off.
It is used to indicate whether the light should turn on when there is an audio / video streaming

Reply:
Yep, that's all it does, and not all that useful IMO.  It would have been a better option to have control over turning the indicator light on/off as a switch and let automation determine if it should turn on for whatever reason - not just when streaming.